### PR TITLE
WScrollBar.paint update

### PR DESCRIPTION
**Pull request by LopyMine**
Updated paint method in WScrollBar widget
Added new textures for WScrollBar
Added ScrollBarTestGui to TestMod

### CODE REVIEW

File: src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java [hunk 1]
```diff
@@ -22,6 +32,28 @@ public class WScrollBar extends WWidget {
...
+			createNinePatch(new Identifier(LibGuiCommon.MOD_ID, "textures/widget/scroll_bar/scroll_bar_pressed.png")),
+			createNinePatch(new Identifier(LibGuiCommon.MOD_ID, "textures/widget/scroll_bar/dark/scroll_bar_pressed.png"))
+	);
+
+	private static final BackgroundPainter scrollBarHovered = BackgroundPainter.createLightDarkVariants(
+			createNinePatch(new Identifier(LibGuiCommon.MOD_ID, "textures/widget/scroll_bar/scroll_bar_hovered.png")),
+			createNinePatch(new Identifier(LibGuiCommon.MOD_ID, "textures/widget/scroll_bar/dark/scroll_bar_hovered.png"))
+	);
+
+	private static final NinePatchBackgroundPainter focus = createNinePatch(new Identifier(LibGuiCommon.MOD_ID, "textures/gui/scroll_bar/scroll_bar_focus.png"));
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java:55]: 1. Static final fields should be in `CONSTANT_CASE`
2. This code will fail to load on a dedi server. You should store these in a `private static final` inner class annotated with `@Environment(CLIENT)` like in `WTabPanel`:
https://github.com/CottonMC/LibGui/blob/e7a36835ff95ac045bbeac156a105c16b4d23df2/src/main/java/io/github/cottonmc/cotton/gui/widget/WTabPanel.java#L442-L456

File: src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java [hunk 2]
```diff
@@ -109,7 +120,7 @@ public boolean canResize() {
 	public boolean canFocus() {
 		return true;
 	}
-
+	@Deprecated
 	private static void drawBeveledOutline(DrawContext context, int x, int y, int width, int height, int topleft, int bottomright) {
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java:124]: Is this method still used somewhere? If yes, it shouldn't be deprecated, and if not, it should be removed.

File: src/testMod/java/io/github/cottonmc/test/client/ScrollBarTestGui.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import io.github.cottonmc.cotton.gui.client.LightweightGuiDescription;
+import io.github.cottonmc.cotton.gui.widget.WPlainPanel;
+import io.github.cottonmc.cotton.gui.widget.WScrollBar;
+import io.github.cottonmc.cotton.gui.widget.data.Axis;
+import io.github.cottonmc.cotton.gui.widget.data.Insets;
+
+import net.fabricmc.fabric.api.util.TriState;
+
+public class ScrollBarTestGui extends LightweightGuiDescription {
+	private boolean darkMode = false;
```
Comment (Juuxel) [inline src/testMod/java/io/github/cottonmc/test/client/ScrollBarTestGui.java:12]: This field is never written to, so I don't really see a point of having it. You could add a toggle button that changes its value, or just rely on the global light/dark mode setting.

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by Juuxel — COMMENTED** [review]
I'd prefer a flat layout for the textures instead of `dark` being a subdirectory so that it'd match the other widgets better, eg:
- `scroll_bar/background_light.png`
- `scroll_bar/background_dark.png`
Additionally, the `scroll_bar_` prefix can be dropped from `background` and `focus`.

(Another naming nit: the `scroll_bar(_pressed|_hovered)` textures should probably use `thumb` as the term since it's used by the scroll bar and slider code. Doesn't really matter though.)

**Comment by LopyMine** [comment]
Okay, I rendered directly via NinePatch.draw, I hope I didn't miss anything.

File: gradle.properties
Comment (Juuxel) [inline gradle.properties:1]: This file shouldn't have the formatting changes

Comment (LopyMine) [reply gradle.properties:1]: wanted to launch minecraft, but to launch it was necessary to update the version of minecraft above, but after that I gave up on this idea) And I accidentally committed this file...

**Comment by LopyMine** [comment]
I made a commit in which I rolled back the gradle.properties file to the original, now everything should be fine

**Comment by LopyMine** [comment]
I remembered here, the user scrolling through the list missed some components, and I would like to have a WScrollBar.setScrollingSpeed(int) method, can I add this method to WScrollBar?

**Comment by Juuxel** [comment]
@LopyMine That is a bit unrelated to this PR so I feel like it could be its own small PR.

**Comment by LopyMine** [comment]
ok, I'll think about adding this as a separate PR later

File: src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java [hunk 4]
```diff
@@ -279,4 +255,19 @@ public void addNarrations(NarrationMessageBuilder builder) {
 		builder.put(NarrationPart.TITLE, NarrationMessages.SCROLL_BAR_TITLE);
 		builder.put(NarrationPart.USAGE, NarrationMessages.SLIDER_USAGE);
 	}
+
+	@Environment(EnvType.CLIENT)
+	final static class Painters {
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java:260]: Nit:
```suggestion
	static final class Painters {
```

File: src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/widget/WScrollBar.java:1]: The formatting changes to unrelated methods should be reverted here to minimise the diff

File: src/testMod/java/io/github/cottonmc/test/client/ScrollBarTestGui.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		root.add(toggleButton, 128 - (toggleButton.getWidth() / 2), 120 - (toggleButton.getHeight() / 2));
+
+		root.validate(this);
+	}
+
+	@Override
+	public TriState isDarkMode() {
+		return TriState.of(darkMode);
+	}
+
```
Comment (Juuxel) [inline src/testMod/java/io/github/cottonmc/test/client/ScrollBarTestGui.java:44]: Nit:
```suggestion
```

**Comment by LopyMine** [comment]
Okay, I fixed it in a new commit.

**Review by Juuxel — APPROVED** [review]
Looks great, thanks for the PR!

**Pull request merged**